services/server/server_windows.go

```diff
@@ -20,12 +20,73 @@ package server
 
 import (
 	"context"
+	"unsafe"
 
+	"github.com/containerd/containerd/log"
 	srvconfig "github.com/containerd/containerd/services/server/config"
 	"github.com/containerd/ttrpc"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
 )
 
-func apply(_ context.Context, _ *srvconfig.Config) error {
+func getPriorityClass(priorityClassName string) uint32 {
+	var priorityClassMap = map[string]uint32{
+		"IDLE_PRIORITY_CLASS":         uint32(windows.IDLE_PRIORITY_CLASS),
+		"BELOW_NORMAL_PRIORITY_CLASS": uint32(windows.BELOW_NORMAL_PRIORITY_CLASS),
+		"NORMAL_PRIORITY_CLASS":       uint32(windows.NORMAL_PRIORITY_CLASS),
+		"ABOVE_NORMAL_PRIORITY_CLASS": uint32(windows.ABOVE_NORMAL_PRIORITY_CLASS),
+		"HIGH_PRIORITY_CLASS":         uint32(windows.HIGH_PRIORITY_CLASS),
+		"REALTIME_PRIORITY_CLASS":     uint32(windows.REALTIME_PRIORITY_CLASS),
+	}
+	return priorityClassMap[priorityClassName]
+}
+
+// addCurrentProcessToJobObjectAndSetPriorityClass creates a new Job Object
+// (https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects),
+// adds the current process to the job object, and specifies the priority
+// class for the job object to the specified value.
+// A job object is used here so that any spawned processes such as CNI or
+// shim binaries are created at the specified thread priority class.
+// Running containerd and shim binaries with above normal / high priority
+// can help improve responsiveness on machines with high CPU utilization.
+func addCurrentProcessToJobObjectAndSetPriorityClass(pc uint32) error {
+	job, err := windows.CreateJobObject(nil, nil)
+	if err != nil {
+		return errors.Wrap(err, "windows.CreateJobObject failed")
+	}
+	limitInfo := windows.JOBOBJECT_BASIC_LIMIT_INFORMATION{
+		LimitFlags:    windows.JOB_OBJECT_LIMIT_PRIORITY_CLASS,
+		PriorityClass: pc,
+	}
+	if _, err := windows.SetInformationJobObject(
+		job,
+		windows.JobObjectBasicLimitInformation,
+		uintptr(unsafe.Pointer(&limitInfo)),
+		uint32(unsafe.Sizeof(limitInfo))); err != nil {
+		return errors.Wrap(err, "windows.SetInformationJobObject failed")
+	}
+	if err := windows.AssignProcessToJobObject(job, windows.CurrentProcess()); err != nil {
+		return errors.Wrap(err, "windows.AssignProcessToJobObject failed")
+	}
+	return nil
+}
```

Comment:
Revisiting this, can we set the JOB_OBJECT_LIMIT_BREAKAWAY_OK flag here? You need to use JOBOBJECT_EXTENDED_LIMIT_INFORMATION in that case but not much of a difference. 
```go
info := windows.JOBOBJECT_EXTENDED_LIMIT_INFORMATION{
		BasicLimitInformation: windows.JOBOBJECT_BASIC_LIMIT_INFORMATION{
			LimitFlags: windows.JOB_OBJECT_LIMIT_BREAKAWAY_OK | windows.JOB_OBJECT_LIMIT_PRIORITY_CLASS,
                        PriorityClass: pc,
	},
}
if _, err := windows.SetInformationJobObject(
	job,
	windows.JobObjectExtendedLimitInformation,
	uintptr(unsafe.Pointer(&info)),
	uint32(unsafe.Sizeof(info))); err != nil {
	return errors.Wrap(err, "windows.SetInformationJobObject failed")
}
```

Reason being the shim will inherit this job object (which we seem to want here) but for any processes that the shim launches we might not want this, for example Host Process containers. We want to launch them in their own job which we can do by just setting the CREATE_BREAKAWAY_FROM_JOB flag on creation of the process which we already do https://github.com/microsoft/hcsshim/blob/ab02b159f6f11b5d91aa4e66afcba9f8f2698919/internal/jobcontainers/jobcontainer.go#L260